### PR TITLE
[CDAP-8440] Bugfixes UI - 3

### PR DESCRIPTION
**Pull request by ajainarayanan**
[CDAP-8459](https://issues.cask.co/browse/CDAP-8459)- Fixes pagination to honor page number in URL
[CDAP-8440](https://issues.cask.co/browse/CDAP-8440)- Fixes all entity cards to have consistent information

- Fixes the bug in Application & Artifact upload wizards, to enable Finish button on file upload

#### Note:
- The bug still exists for Plugin Upload wizard as we use the same PluginUploadStore for both the steps.  But nevertheless its a not a blocker bug. Hence deferring it for a separate PR.

### CODE REVIEW

**Comment by bdmogal** [comment]
@ajainarayanan when I hit an out-of-bounds page (say page 6 when there are only 4 pages), I get an improper error message -

image


Perhaps it should say "Page 5 does not exist. Please go back to an earlier page."

File: cdap-ui/app/cdap/components/EntityListView/index.js [hunk 1]
```diff
@@ -270,10 +270,10 @@ class EntityListView extends Component {
         } else {
           this.setState({
             notFound: false,
-            currentPage: 1
+             currentPage: this.state.currentPage
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/EntityListView/index.js]: nit: extra tab

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by ajainarayanan** [comment]
Bamboo build: http://builds.cask.co/browse/CDAP-DRC5692/latest

**Pull request merged**